batch7crud-roll6/src/main/java/com/lftechnology/batch7crud/controller/UserController.java

```diff
@@ -0,0 +1,170 @@
+ */
+
+@WebServlet("/users/*")
+public class UserController extends HttpServlet {
+    private static final long serialVersionUID = 1L;
+    private UserService userService = new UserService();
+
+    /**
+     * @see HttpServlet#HttpServlet()
+     */
+    public UserController() {
+        super();
+        // TODO Auto-generated constructor stub
+    }
+
+    /**
+     * @see HttpServlet#doGet(HttpServletRequest request, HttpServletResponse response)
+     */
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        String urlString = request.getPathInfo();
+
+        if (urlString == null) {
+            list(request, response, 1);
+
+        } else {
+
+            String[] pathArgs = urlString.split("/");
+
+            if (pathArgs[0].equals("") && pathArgs[1].equals("add")) {
+                create(request, response);
+            } else if (pathArgs[0].equals("") && pathArgs[2].equals("edit")) {
+                int userID = TypeCaster.toInt(pathArgs[1]);
+                edit(request, response, userID);
+            }
+
+        }
+
+    }
+
+    /**
+     * @see HttpServlet#doPost(HttpServletRequest request, HttpServletResponse response)
+     */
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+
+        String urlString = request.getPathInfo();
+        if (urlString == null) {
+            list(request, response, 1);
+        } else {
+            String[] pathArgs = urlString.split("/");
+            if (pathArgs[0].equals("") && pathArgs[1].equals("add")) {
+                createProcess(request, response);
+            } else if (pathArgs[0].equals("") && pathArgs[2].equals("edit")) {
+                int userID = TypeCaster.toInt(pathArgs[1]);
+                editProcess(request, response, userID);
+            } else if (pathArgs[0].equals("") && pathArgs[2].equals("delete")) {
+                int userID = TypeCaster.toInt(pathArgs[1]);
+                deleteProcess(request, response, userID);
+            }
+
+        }
+
+    }
+
+    private void list(HttpServletRequest request, HttpServletResponse response, int page) throws ServletException, IOException {
+        try {
+            request.setAttribute("users", userService.fetch(1));
+            request.getRequestDispatcher("/WEB-INF/views/listUser.jsp").forward(request, response);
+
+        } catch (DataException e) {
+            e.printStackTrace();
+
+        }
+    }
+
+    private void create(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        request.getRequestDispatcher("/WEB-INF/views/addUser.jsp").forward(request, response);
+    }
+
+    private void createProcess(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+
+        User user = new User();
+        UserService userService = new UserService();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename "userService" which hides the field declared at line 24. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AHiddenFieldCheck)